src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTime.cs

```diff
@@ -269,17 +269,28 @@ private void WriteStringValue(DateTime value, ref int idx)
             _buffer[idx++] = JsonConstants.Quote;
         }
 
-        private static readonly StandardFormat s_dateTimeStandardFormat = new StandardFormat('O');
-
         private void FormatLoop(DateTime value, ref int idx)
         {
-            if (!Utf8Formatter.TryFormat(value, _buffer.Slice(idx), out int bytesWritten, s_dateTimeStandardFormat))
+            Span<byte> tempSpan = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            bool result = Utf8Formatter.TryFormat(value, tempSpan, out int bytesWritten, s_dateTimeStandardFormat);
+            Debug.Assert(result);
+
+            JsonWriterHelper.TrimDateTimeOffsetString(tempSpan.Slice(0, bytesWritten), out bytesWritten);
+
+            if (_buffer.Slice(idx).Length < bytesWritten)
```

Comment:
nit: Update `tempSpan` with the slice since its common between both branches.

`tempSpan = tempSpan.Slice(0, bytesWritten)`

or maybe:

```C#
if (_buffer.Slice(idx).Length < bytesWritten)
{
   AdvanceAndGrow(ref idx, bytesWritten);
}

tempSpan.Slice(0, bytesWritten).CopyTo(_buffer.Slice(idx));
```